sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1015,8 +1016,15 @@ def test_neighbors_metrics(n_samples=20, n_features=3,
                                                algorithm=algorithm,
                                                metric=metric, p=p,
                                                metric_params=metric_params)
-            neigh.fit(X)
-            results[algorithm] = neigh.kneighbors(test, return_distance=True)
+            if metric == 'haversine':
+                neigh.fit(X[:, :2])
```

Comment:
to make this clearer, you could instead do `feature_sl = slice(None, 2) if metric == 'haversine' else slice(None)` and use that slice object.

Reply:
Ready